### ISSUE TITLE
Per Plugin metrics values are not properly displayed in _node/stats/pipeline API

### ISSUE THREAD

**Issue by suyograo** [elastic/logstash#5529]
Tried to add number of matches and failures to grok filter.

```
curl localhost:9600/_node/stats/pipeline?pretty
```

Response:
```
{
        "id" : "geoip_a70e162f-1789-4526-ae2b-b6eae7655d2f",
        "events" : {
          "duration_in_millis" : 78,
          "in" : 1,
          "out" : 1
        },
        "name" : "geoip"
      }, {
        "id" : "matches",
        "namespaces" : [ "stats", "pipelines", "main", "plugins", "filters" ],
        "key" : "matches",
        "type" : "counter",
        "value" : 1
      }, {
```

I expected 

```
"matches": 1
```

**Comment by ph** [comment]
By default it will pickup `to_hash` of the base metric so this why you get all the unwanted data from the custom metric.

**Comment by suyograo** [comment]
Fixed in #5532
